### PR TITLE
Implement logs

### PR DESCRIPTION
**Pull request by chestnutcone**
Added loggers, but didnt really log anything because didn't know what to log. I think some loggings will occur in views, which have to wait till front end and back end integration to see what might possibly go wrong. Added holiday notification to function, but did not modify views. Added test cases to accommodate for holiday

### CODE REVIEW

File: schedule/tests/test_models.py [hunk 1]
```diff
@@ -183,18 +202,41 @@ def testing_behavior(self, start_date, diff, employees, shift_pattern,
         start_date = datetime.date(2019, 9, 4)  # wednesday
         diff = datetime.timedelta(days=3)
         expected_status = [False, False, False]
-        expected_not_registered = [[(datetime.date(2019, 9, 4), night_shift)],
-                                   [(datetime.date(2019, 9, 7), night_shift),
-                                    (datetime.date(2019, 9, 8), night_shift)],
-                                   [(datetime.date(2019, 9, 10), night_shift)]]
+        expected_status_detail = [{'overridable': [],
```
Comment (coolgary100) [inline schedule/tests/test_models.py]: Not sure if you could just create a method to construct some of these larger variables in your test cases, this way you don't need to keep copy pasting the declaration and it'll be easier to add future test cases.

File: project_specific/migrations/0001_initial.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Generated by Django 2.2.1 on 2019-09-24 23:32
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    initial = True
+
+    dependencies = [
```
[the remark this answers is not in the document]
Comment (chestnutcone) [reply project_specific/migrations/0001_initial.py]: eh no, all migrations are auto generated

File: project_specific/migrations/0001_initial.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name='Organization',
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('name', models.CharField(max_length=100)),
```
[the remark this answers is not in the document]
Comment (chestnutcone) [reply project_specific/migrations/0001_initial.py]: lmao it was arbitrary. When i was writing this, i was thinking about the troll longest name

File: schedule/models.py [hunk 4]
```diff
@@ -501,19 +548,14 @@ def swap(person, swap_shift_start):
...
-            else:
-                print('no people are available to swap')
+            if not free_people:
                 success = False
 
                 # go into queue of holding, wait till database updates, then run rechecks
     return {'success': success,
             'available_shifts': output,
-            'free_people': free_people}
+            'free_people': free_people,
```
[the remark this answers is not in the document]
Comment (chestnutcone) [reply schedule/models.py]: yeah i know, but i dont know what else I can name it..need help

File: schedule/models.py [hunk 5]
```diff
@@ -180,13 +196,30 @@ def get_schedule(person):
     return existing_schedule
 
 
-def set_schedule_day(person, start_day, shift):
+def set_schedule_day(person, start_day, shift, override=False):
     """
     set schedule for a single day on that person
```
[the remark this answers is not in the document]
Comment (chestnutcone) [reply schedule/models.py]: both option you wrote looks the same to me?

File: schedule/models.py [hunk 6]
```diff
@@ -242,51 +276,71 @@ def set_schedule_day(person, start_day, shift):
...
-def set_schedule(person, start_date, shift_pattern, repeat=1):
+def set_schedule(person, start_date, shift_pattern, repeat=1, override=False):
     """
     This function sets a person's schedule
     
     person is Employee instance
     start_date is datetime.date object
     shift is shift pattern from Schedule
     """
+    status_detail = {'overridable': [],
```
[the remark this answers is not in the document]
Comment (chestnutcone) [reply schedule/models.py]: ok got it

File: user/views.py [hunk 7]
```diff
@@ -6,6 +6,22 @@
...
+import logging
+import os
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+if not logger.handlers:
+    log_filename = 'logs/user_views.log'
+    os.makedirs(os.path.dirname(log_filename), exist_ok=True)
+
```
[the remark this answers is not in the document]
Comment (chestnutcone) [reply user/views.py]: do you mean to make another variable for os.path.dirname(log_filename) ?

File: user/views.py [hunk 8]
```diff
@@ -79,7 +93,7 @@ def post(self, request, *args, **kwargs):
                             try:
                                 model_add_perm = Permission.objects.get(name=name)
                             except Permission.DoesNotExist:
-                                print('Permission not found with name:', name)
+                                logger.exception('Permission not found with name:', name)
```
[the remark this answers is not in the document]
Comment (chestnutcone) [reply user/views.py]: i think 'except' has the same meaning as catch? There are lots of error built in by python you can do with except. Ie IndexError, ImportError etc

File: schedule/tests/test_models.py (hunk 1 above)
Comment (chestnutcone) [reply schedule/tests/test_models.py]: but the method that needs to create those is the method im testing

**Pull request merged**